File: python/reverse_bits.py
class Solution:
    # @param n, an integer
    # @return an integer
    # O(1) time | O(1) space
    def reverseBits(self, n):
        res = 0
        for i in range(32):
            bit = (n >> i) & 1
            res += (bit << (31 - i))
        return res

File: src/main/java/april_2025/ReverseBits.java
package april_2025;

public class ReverseBits {

    // O(1) time | O(1) space
    public int reverseBits(int n) {
        int res = 0;

        for (int i = 0; i < 32; i++) {
            int bit = (n >> i) & 1;
            res = res | (bit << (31 - i));
        }
        return res;
    }

}
